x/ibc/02-client/keeper/keeper.go

```diff
@@ -197,6 +200,47 @@ func (k Keeper) GetSelfConsensusState(ctx sdk.Context, height uint64) (exported.
 	return consensusState, true
 }
 
+// ValidateClient validates the client parameters for a client of the running chain
+func (k Keeper) ValidateClient(ctx sdk.Context, clientState exported.ClientState) error {
```

Comment:
I'd prefer to move this into the client state to allow more flexibility in the implementation:

`clientState.ValidateClient(height int64)`

Reply:
or write it in a way that supports other client states

Reply:
I totally agree. We should not introduce notions of tendermint clients into `02-client`. This also prevent the introduction of any other client including solo machine

Reply:
> I see this should be validating the client state the counterparty stores for us

Correct.

> I still think we should try to work towards making the SDK not tendermint dependent. And making this a client state func would help that

This seems like premature optimization to me, given that even basic things like the sdk.Context assume a Tendermint backend. Also, this never needs to be part of the interface of a ClientState. Since this will always be an SDK function checking it's own client, it doesn't need to be exposed or generalized beyond that. While SDK chains remain Tendermint chains, it can do what I've done above.

If SDK ever generalizes beyond Tendermint, this function can be modified within that SDK implementation to be equally generalized as well WITHOUT ever having to add this function to any interface.

Reply:
> This seems like premature optimization to me, given that even basic things like the sdk.Context assume a Tendermint backend. Also, this never needs to be part of the interface of a ClientState. Since this will always be an SDK function checking it's own client, it doesn't need to be exposed or generalized beyond that. While SDK chains remain Tendermint chains, it can do what I've done above.

I agree with the argument you stated above. Although I would prefer for this to be a wrapper function:

```
func (k Keeper) ValidateClient(ctx sdk.Context, clientState exported.ClientState) error {
	switch clientState.ClientType() {
	case exported.Tendermint:
		return tendermint.ValidateClient(ctxm clientState)
	default:
	// error
	}


```